lib/handlers/quote-to-ratio/quote-to-ratio.ts

```diff
@@ -0,0 +1,422 @@
+      quote,
+      quoteGasAdjusted,
+      route,
+      optimalRatio,
+      postSwapTargetPool,
+      trade,
+      estimatedGasUsed,
+      estimatedGasUsedQuoteToken,
+      estimatedGasUsedUSD,
+      gasPriceWei,
+      methodParameters,
+      blockNumber,
+    } = swapRoute;
+
+    const routeResponse: Array<PoolInRoute[]> = [];
+
+    for (const subRoute of route) {
+      const {
+        route: { tokenPath, pools },
+        amount,
+        quote,
+      } = subRoute;
+
+      const curRoute: PoolInRoute[] = [];
+      for (let i = 0; i < pools.length; i++) {
+        const nextPool = pools[i];
+        const tokenIn = tokenPath[i];
+        const tokenOut = tokenPath[i + 1];
+
+        let edgeAmountIn = undefined;
+        if (i == 0) {
+          edgeAmountIn =
+            type == 'exactIn'
+              ? amount.quotient.toString()
+              : quote.quotient.toString();
+        }
+
+        let edgeAmountOut = undefined;
+        if (i == pools.length - 1) {
+          edgeAmountOut =
+            type == 'exactIn'
+              ? quote.quotient.toString()
+              : amount.quotient.toString();
+        }
+
+        curRoute.push({
+          type: 'v3-pool',
+          address: poolProvider.getPoolAddress(
+            nextPool.token0,
+            nextPool.token1,
+            nextPool.fee
+          ).poolAddress,
+          tokenIn: {
+            chainId: tokenIn.chainId,
+            decimals: tokenIn.decimals.toString(),
+            address: tokenIn.address,
+            symbol: tokenIn.symbol!,
+          },
+          tokenOut: {
+            chainId: tokenOut.chainId,
+            decimals: tokenOut.decimals.toString(),
+            address: tokenOut.address,
+            symbol: tokenOut.symbol!,
+          },
+          fee: nextPool.fee.toString(),
+          liquidity: nextPool.liquidity.toString(),
+          sqrtRatioX96: nextPool.sqrtRatioX96.toString(),
+          tickCurrent: nextPool.tickCurrent.toString(),
+          amountIn: edgeAmountIn,
+          amountOut: edgeAmountOut,
+        });
+      }
+
+      routeResponse.push(curRoute);
+    }
+
+    const tokenIn = trade.inputAmount.currency.wrapped
+    const tokenOut = trade.outputAmount.currency.wrapped
+
+    const zeroForOne = tokenIn == token0
+    let token0BalanceUpdated: CurrencyAmount<Currency>;
+    let token1BalanceUpdated: CurrencyAmount<Currency>;
+    let optimalRatioAdjusted: Fraction;
+    let optimalRatioDecimal: string;
+    let newRatioDecimal: string
+    if (zeroForOne) {
+      token0BalanceUpdated = token0Balance.subtract(trade.inputAmount)
+      token1BalanceUpdated = token1Balance.add(trade.outputAmount)
+      optimalRatioAdjusted = optimalRatio
+      optimalRatioDecimal = optimalRatioAdjusted.toFixed(token0.wrapped.decimals)
+      newRatioDecimal = new Fraction(
+        token0BalanceUpdated.quotient.toString(),
+        token1BalanceUpdated.quotient.toString()
+      ).toFixed(token0.wrapped.decimals)
```

Comment:
since this is in terms of x token0 for token1, I do the decimal in terms of token0. Since the numbers on chain tend to be multiplied by e18, I thought it might be useful to keep the precision relatively big